Data_Collection/CRON_scripts/colombia_data.py

```diff
@@ -14,83 +14,103 @@
+            client.get(
+                "gt2j-8ykr",
+                offset=start,
+                limit=chunk_size,
+                order="fecha_reporte_web",
+                where=f'fecha_reporte_web > "{slice_date}"',
+            )
+        )
+
+        df_new = df_new.rename(columns=str.lower)
 
-        df_new = df_new.rename(columns = str.lower)
-        
         if df_new.empty:
-            break 
-        
-        df_new.index.name = 'id_'
-        df_new.reset_index(inplace = True)
-        df_new.set_index(['id_', 'id_de_caso'] , inplace = True)
-                
+            break
+
+        df_new.index.name = "id_"
+        df_new.reset_index(inplace=True)
+        df_new.set_index(["id_", "id_de_caso"], inplace=True)
+
         df_new = df_new.convert_dtypes()
-        
-        # transform the datetime columns in the correct time 
+
+        # transform the datetime columns in the correct time
         for c in df_new.columns:
-            if c.lower().startswith('fecha'):
-                df_new[c] = pd.to_datetime(df_new[c], errors='coerce')
+            if c.lower().startswith("fecha"):
+                df_new[c] = pd.to_datetime(df_new[c], errors="coerce")
 
         # eliminate any space in the end and start of the string values
-        for i in df_new.select_dtypes(include=['string']).columns:
+        for i in df_new.select_dtypes(include=["string"]).columns:
             df_new[i] = df_new[i].str.strip()
 
         # Move up the starting record
         start = start + chunk_size
-        
+
         yield df_new
-        
+
 
 def load_into_db(client):
-        
+
     slice_date = datetime.date(datetime.today()) - timedelta(200)
-    slice_date = slice_date.strftime('%Y-%m-%d')
+    slice_date = slice_date.strftime("%Y-%m-%d")
+
+    # count the number of records that will be fetched
+    records = client.get_all(
+        "gt2j-8ykr", select="COUNT(*)", where=f'fecha_reporte_web > "{slice_date}"'
+    )
 
-    # count the number of records that will be fetched 
-    records = client.get_all("gt2j-8ykr", select = 'COUNT(*)', where = f'fecha_reporte_web > "{slice_date}"')
-    
     for i in records:
         record_count = i
         break
-    
+
     del records
-    
+
     start = 0
     chunk_size = 10000
-    maxrecords = int(record_count['COUNT'])
-    
-    engine = create_engine('postgresql://epigraph:epigraph@localhost:5432/epigraphhub')
-    
+    maxrecords = int(record_count["COUNT"])
+
+    engine = create_engine(config.DB_URI)
+
     for df_new in chunked_fetch(start, chunk_size, maxrecords):
-   
+
         # save the data
         with engine.connect() as conn:
-            upsert(con=conn, df = df_new, table_name='positive_cases_covid_d', schema='colombia', if_row_exists= 'update',
-                chunksize=1000, add_new_columns=True, create_table= False) 
-            
-    logger.info('table casos_positivos_covid updated')
-                
-            
+            upsert(
```

Comment:
If the "on conflict update" is being done by Pangres upsert function, we need to know how it is doing it or switch to doing it ourselves using plain SQL.